src/Form/LeaseBookType.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace App\Form;
+
+use App\Entity\Book;
+use App\Entity\Reader;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\DateType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormError;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Class LeaseBookType
+ * Used to handle the book leasing process
+ *
+ * @package App\Form
+ */
+class LeaseBookType extends AbstractType
+{
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('reader', EntityType::class, [
+                'class' => Reader::class,
+                'choice_label' => 'name',
+            ])
+            ->add('returnDate', DateType::class, [
+                'widget' => 'single_text',
+            ])
+        ;
+
+        /**
+         * Check - before data to be set to the object - if return date is higher than the current date,
+         * the book is not available to be leased
+         */
+        $builder->addEventListener(FormEvents::POST_SUBMIT, function (FormEvent $event) {
```

Comment:
Logic like this belongs in the entity itself - it should decide, how to react to an invalid action, or determine the validity of the command. 
If it breaks here, it will take long time to debug it.

Reply:
Yeap, you're right . I was like very under pressure and maybe that's why I implemented this validation solution ...